Add sub type name to user-identity struct

andyland opened the pull request.

andyland committed:
Add sub type name to user-identity struct

andyland committed:
Add sub type name to user-identity struct

eggman64 reviewed (changes requested):
Hi @andyland!

Thanks for contributing. I looked through the Zendesk API [here](https://developer.zendesk.com/rest_api/docs/support/user_identities) and couldn't find a key matching `subtype_name`. Could you point me to an explanation/usage of this in the official Zendesk docs?
Thanks once again

andyland commented:
@eggman64 Yeah, that's odd, it appears to be undocumented.  I received information about the property through developer support.  For our use case, it's used for identifying the Apple Business Chat id and has the value of `Apple Business Chat`.  It appears to be omitted in the other channels we use.

I will reach out to them, ask that they update their official documentation and report back.

edevil committed:
Support untyped (Additional) Collaborators & Requesters (#76)

* Support untyped (Additional)Collaborators

* Also support the requester attribute

* Use pointer/omitempty convention

edevil committed:
Added List Tickets by External ID (#77)

https://developer.zendesk.com/rest_api/docs/support/tickets#list-tickets-by-external-id

edevil committed:
New function to list comments that exposes options (#78)

* New function to list comments that exposes options
https://developer.zendesk.com/rest_api/docs/support/ticket_comments#list-email-ccs-for-a-comment

* Added comment with link to API docs

edevil committed:
Added more ticket-related methods (#80)

* Added more ticket-related methods

* Added list tickets method

FmeuMan committed:
List audits by ticket (#79)

hypnoglow committed:
Init go module (#83)

hypnoglow committed:
Use ClientOption to set http client (#84)

fterrag committed:
Adds CreateGroup, DeleteGroup, and ListGroups to zendesk.Client (#87)

* Adds CreateGroup, DeleteGroup, and ListGroups to zendesk.Client

* Remove spew from go.mod

Wundark committed:
Add Make Identity Primary method (#89)

Wundark committed:
Add Show Many Organizations endpoint (#90) (#91)

Wundark committed:
Add Show Many Users by External IDs (#92) (#93)

jinmatt committed:
Exclude .vscode config

jinmatt committed:
Lists schedules

jinmatt committed:
Merge branch 'master' of github.com:gomaps/go-zendesk

jinmatt committed:
Merge branch 'jinmatt/schedule'

andyland committed:
Add incremental users endpoint

jinmatt committed:
Updates SoruceInfo type

andyland committed:
Change module path

jinmatt committed:
Updated dependencies go.sum
